### DISCUSSION TITLE
Image Component with AWS Throws 400 Errors in Production

### DISCUSSION THREAD

**Discussion by AlexandraKlein** [vercel/next.js#19089] [Help]
# Bug report

## Describe the bug

Implementing Next Image works locally, but throws 400 errors in Production with `"url" parameter is required` response.

### next.config.js
```javascript
module.exports = {
  images: {
    domains: [
      'images.ctfassets.net'
    ]
  }
}
```

### Implementaion
```javascript
<Image
  src={`https:${image.fields.file.url}`}
  alt={image.fields.title}
  width={image.fields.file.details.image.width}
  height={image.fields.file.details.image.height}
/>
```

Src url in production with 400 error: `https://mywebsite.com/_next/image?url=https%3A%2F%2Fimages.ctfassets.net%2F1xkvhwkgcr51%2F6HODBVSFqUTbEKI7ilXx%2Ff9f16ab3552e973b764c922f06db181a%2Fmy-image.png&w=1200&q=75`

Src url locally with no error/displays fine: `http://localhost:3000/_next/image?url=https%3A%2F%2Fimages.ctfassets.net%2F1xkvhwkgcr51%2F6HODBVSFqUTbEKI7ilXx%2Ff9f16ab3552e973b764c922f06db181a%2Fmy-image.png&w=1200&q=75`

- OS: macOS Catalina 10.15.7
- Version of Next.js: 10.0.1
- Version of Node.js: v15.1.0

**Comment by styfle** [comment]
What is "production" in this case?

Does it work with `next build && next start` visiting localhost?

**Comment by AlexandraKlein** [comment]
@styfle  Yes, it works with `next build && next start`. Production is on the server hosted by AWS. Please see the src url's posted in the description.

**Comment by styfle** [comment]
It sounds like you have a proxy that isn't forwarding the query string parameters

**Comment by AlexandraKlein** [comment]
I think once https://github.com/vercel/next.js/pull/19117 merges, will be fixed with contentful loader.

**Comment by ynotdraw** [comment]
Seeing something extremely similar, where locally all is fine; however, once deployed to an environment via CI, getting 400-level errors.

**Reply by AlexandraKlein** [reply]
Looks like we have to wait on https://github.com/vercel/next.js/pull/20216

**Comment by daniel-keller** [comment]
I'm not sure if this is related but I've had a similar issue where images hosted locally with the site throw 403 errors. 403 errors are inconsistently thrown if a user scrolls through the page too fast and the nextjs `<Image/>` component tries to lazy load the image (only happens on the server, not in a dev environment).

**Comment by sohhamm** [comment]
how did you resolve this facing the same issue right now

**Reply by AlexandraKlein** [reply]
My issue was with Contentful. I resolved with a custom image loader: 

```javascript
const contentfulLoader = ({ src, quality, width }) => {
  const params = [`w=${width}`];

  if (quality) {
    params.push(`q=${quality}`);
  }

  return `${src}?${params.join('&')}`;
};

...

 <Image
    src={image.fields.file.url}
    loader={contentfulLoader}
    ...
  />
```
